Sources/UI/SwiftUI/Views/AnimatedImage/AnimatedImage.swift

```diff
@@ -9,28 +9,107 @@
 import Kingfisher
 import SwiftUI
 
-/// A view that displays an animated image either from a local file or a remote URL.
+/// A view that displays an animated image (a.k.a GIF) either from a local file or a remote URL.
 ///
 /// It uses `Kingfisher` for handling the image loading from both sources.
+/// This implementation provides access to advanced animation customization options.
 @available(iOS 15.0, *)
 public struct AnimatedImage: View {
   private let source: Source
   private let animated: Bool
+  private let autoStartAnimation: Bool
+  private let repeatCount: AnimatedImageView.RepeatCount
+  private let onAnimationStart: (() -> Void)?
+  private let onAnimationEnd: (() -> Void)?
   
-  public init(source: Source, animated: Bool = false) {
+  public init(
+    source: Source, 
+    animated: Bool = false,
+    autoStartAnimation: Bool = true,
+    repeatCount: AnimatedImageView.RepeatCount = .infinite,
+    onAnimationStart: (() -> Void)? = nil,
+    onAnimationEnd: (() -> Void)? = nil
+  ) {
     self.source = source
     self.animated = animated
+    self.autoStartAnimation = autoStartAnimation
+    self.repeatCount = repeatCount
+    self.onAnimationStart = onAnimationStart
+    self.onAnimationEnd = onAnimationEnd
   }
   
   public var body: some View {
+    AnimatedImageViewRepresentable(
+      source: source,
+      animated: animated,
+      autoStartAnimation: autoStartAnimation,
+      repeatCount: repeatCount,
+      onAnimationStart: onAnimationStart,
+      onAnimationEnd: onAnimationEnd
+    )
+  }
+}
+
+@available(iOS 15.0, *)
+private struct AnimatedImageViewRepresentable: UIViewRepresentable {
+  let source: AnimatedImage.Source
+  let animated: Bool
+  let autoStartAnimation: Bool
+  let repeatCount: AnimatedImageView.RepeatCount
+  let onAnimationStart: (() -> Void)?
+  let onAnimationEnd: (() -> Void)?
+  
+  func makeUIView(context: Context) -> AnimatedImageView {
+    let imageView = AnimatedImageView()
+    imageView.autoPlayAnimatedImage = autoStartAnimation
+    imageView.repeatCount = repeatCount
+    imageView.delegate = context.coordinator
+    return imageView
+  }
+  
+  func updateUIView(_ imageView: AnimatedImageView, context: Context) {
+    // Update the image source
     switch source {
     case .local(let fileName):
       if let fileUrl = Bundle.main.url(forResource: fileName, withExtension: "gif") {
-        KFAnimatedImage(source: .provider(LocalFileImageDataProvider(fileURL: fileUrl)))
-          .fade(duration: animated ? 0.25 : 0)
+        imageView.kf.setImage(
+          with: .provider(LocalFileImageDataProvider(fileURL: fileUrl)),
+          options: [
+            .transition(.fade(animated ? 0.25 : 0))
```

Comment:
Is this animation something that should be controlled by the user instead?

Reply:
Good point. Let me think about it.

Reply:
I've landed up [opening all the options](https://github.com/povio/PovioKit/pull/355/commits/47e9a8a568fbc8dda0340fcea418f2daadd7c413) available to be customized.

Reply:
Lmk how does that feel.